src/app/components/SubjectHeading/SubjectHeadingShow.jsx

```diff
@@ -171,7 +172,7 @@ class SubjectHeadingShow extends React.Component {
                 }
               />
             </div>
-            : null
+            : <LocalLoadingLayer message="Loading More Subject Headings" />
```

Comment:
What if `contextHeadings` is an empty array or something else false-y for some reason?  The local loading layer would misleadingly appear. The pattern used for `BibsList` is more accurate.

Reply:
Empty array is strangely truthy. But point taken that we need to ensure `contextHeadings` is an array. I've added some extra validations/error handling which should make sure `contextHeadings` always resolves to an array after the initial API call. Let me know what you think. We can always go with the `BibsList` pattern but I'd like to keep this component stateless if possible.

Reply:
I do think the logic is a little funny and misses the point. What if only the `context` query times out or has some other error? We could still load the bibs and the related headings. Currently, none of the components would load.

Reply:
Also, this component would handle the `contextLoading` logic still. `BibsList` is a class component and managed the local loading logic, because currently it makes the bibs API call.

Reply:
Sorry, I don't think I follow. Why would the bibs and related headings not load if the `context` query times out?

Reply:
On line 135, if the error state is true, only an error message is rendered.

Reply:
OK now I understand what you meant about this being a pre-existing issue. Good spot.